gunicorn/arbiter.py

```diff
@@ -460,10 +461,12 @@ def reap_workers(self):
                     if exitcode == self.APP_LOAD_ERROR:
                         reason = "App failed to load."
                         raise HaltServer(reason, self.APP_LOAD_ERROR)
-                    worker = self.WORKERS.pop(wpid, None)
-                    if not worker:
-                        continue
-                    worker.tmp.close()
+
+                    worker = (self.WORKERS.pop(wpid, None)
```

Comment:
I would add a comment here to say to explain that this function might happen before the SIGCHLD handler for this pid.